Xamarin.Forms.Platform.iOS/Platform.cs

```diff
@@ -303,100 +303,180 @@ bool PageIsChildOfPlatform(Page page)
 			return Page == page || _modals.Contains(page);
 		}
 
-		// Creates a UIAlertAction which includes a call to hide the presenting UIWindow at the end
-		UIAlertAction CreateActionWithWindowHide(string text, UIAlertActionStyle style, Action setResult, UIWindow window)
+		void PresentAlert(AlertArguments arguments)
 		{
-			return UIAlertAction.Create(text, style,
-					a =>
+			PresentPopUp(false, (window, onComplete) =>
+			{
+				UIViewController alertViewController = null;
+
+				if (arguments.Visual.IsMaterial())
```

Comment:
My first thought about this is that we probably need to think of a way to make a hook for this so people can provide there own implementations.  The idea with Visual is that anyone can hook in a "Visual" so we need a way of asking for an implementation based on the Visual. Maybe something that's registered into Dependency Service? 

Which also plays into your comment
*// TODO: move this out of the main assembly for linking*

Because we'll need some mechanism for that right?

Reply:
Though we can probably get that implemented later on so that we can merge this PR

Reply:
This is usually done through the IRegisterable interface and then creating a connection between two types. The cross platform and then the native

I was thinking maybe we create a new empty class that's just something like
*Class Alertview* or we anchor the registerable to *ActionSheetArguments* which is what is passed to *Internals.Registrar.Registered.GetHandler* to retrieve some interface that's used to pass the arguments to

Reply:
Yeah. I think we are on the same page here. We need to have some "registerable" something that can be looked up for "system" concepts - such as alerts or, if we ever go this route, local notifications. Not quite "renderers", more "logic chunks".

Reply:
Yea this is how things like the ImageHandlers work. They aren't renderers they just get registered against the ImageSources like

https://github.com/xamarin/Xamarin.Forms/blob/release-3.4.0-sr1/Xamarin.Forms.Platform.Android/Properties/AssemblyInfo.cs#L64

So I'm thinking something similar to that